library/nxos_get_vlan.py

```diff
@@ -0,0 +1,216 @@
+    new_dict = {}
+    for key, value in table.items():
+        new_key = key_map.get(key)
+        if new_key:
+            value = table.get(key)
+            if value:
+                new_dict[new_key] = str(value)
+            else:
+                new_dict[new_key] = value
+    return new_dict
+
+
+def get_vlan_facts(body):
+    vlan_facts = []
+    vlan_table = body['TABLE_vlanbriefxbrief']['ROW_vlanbriefxbrief']
+
+    key_map = {
+                "vlanshowbr-vlanid-utf": "vlan_id",
+                "vlanshowbr-vlanname": "name",
+                "vlanshowbr-shutstate": "admin_state",
+                "vlanshowbr-vlanstate": "state",
+                "vlanshowplist-ifidx": "interfaces"
+            }
+
+    if isinstance(vlan_table, dict):
+        vlan_table = [vlan_table]
+
+    for each in vlan_table:
+        mapped_vlan_facts = apply_key_map(key_map, each)
+        try:
+            if mapped_vlan_facts['interfaces']:
+                mapped_vlan_facts['interfaces'] = mapped_vlan_facts['interfaces'].split(',')
+        except KeyError:
+            mapped_vlan_facts['interfaces'] = []
+        vlan_facts.append(mapped_vlan_facts)
+    return vlan_facts
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            vlan_id=dict(required=False, default='all'),
+            protocol=dict(choices=['http', 'https'], default='http'),
+            port=dict(required=False, type='int', default=None),
+            host=dict(required=True),
+            username=dict(type='str'),
+            password=dict(type='str'),
+        ),
+        supports_check_mode=False
+    )
+    if not HAS_PYCSCO:
+        module.fail_json(msg='There was a problem loading pycsco')
+
+    auth = Auth(vendor='cisco', model='nexus')
+    username = module.params['username'] or auth.username
+    password = module.params['password'] or auth.password
+    protocol = module.params['protocol']
+    port = module.params['port']
+    host = socket.gethostbyname(module.params['host'])
+
+    vlan_id = module.params['vlan_id']
+
+    try:
+        vlan_id = int(vlan_id)
+    except ValueError:
+        if vlan_id != 'all':
+            module.fail_json(msg='The value {0} is not a '
+                                 'valid Vlan ID. This should be '
+                                 'an integer or "all".'.format(vlan_id))
+
+    device = Device(ip=host, username=username, password=password,
+                    protocol=protocol, port=port)
+
+    show_vlan_command = 'show vlan brief'
+
+    show_vlan_body = parsed_data_from_device(device, show_vlan_command, module)
+    vlan_facts = get_vlan_facts(show_vlan_body)
+
+    if isinstance(vlan_id, int):
+        present = False
+        for each in vlan_facts:
+            if int(each['vlan_id']) == vlan_id:
+                present = True
+                vlan_facts = each
+        if not present:
+            vlan_facts = {}
+
+    module.exit_json(vlan_facts=vlan_facts)
```

Comment:
Let's change to `module.exit_json(resource=vlan_data)`  and change vlan_facts to vlan_data where needed.